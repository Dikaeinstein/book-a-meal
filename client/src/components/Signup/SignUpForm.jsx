import React from 'react';
import PropTypes from 'prop-types';
import { Form, Field } from 'formik';
import { ProgressBar } from 'react-materialize';
import Label from '../util/Label';
import Button from '../util/Button';
import Loading from '../util/Loading';

const SignUpForm = (props) => {
  const {
    errors,
    touched,
    isSubmitting,
  } = props;
  return (
    <Form className="form card">
      <h2 className="text-center">Sign Up</h2>
      <p className="text-center font-weight-bold">
        Sign up to book a meal now
      </p>
      <div>
        <Label className="label label-block" htmlFor="name">
          Full name
        </Label>
        <Field
          placeholder="Enter full name"
          id="name"
          name="name"
          required
          className={touched.name
          && errors.name ? 'input-error' : ''}
        />
        {
          touched.name
          && errors.name
          && <div className="error">{errors.name}</div>
        }
      </div>
      <div>
        <Label className="label label-block" htmlFor="email">
          Email address:
        </Label>
        <Field
          type="email"
          placeholder="johndoe@domain.com"
          id="email"
          name="email"
          required
          className={touched.email
          && errors.email ? 'input-error' : ''}
        />
        {
          touched.email
          && errors.email
          && <div className="error">{errors.email}</div>
        }
      </div>
      <div>
        <Label className="label label-block" htmlFor="password">
          Password:
        </Label>
        <Field
          type="password"
          placeholder="Enter password"
          id="password"
          name="password"
          required
          className={touched.password
          && errors.password ? 'input-error' : ''}
        />
        {
          touched.password
          && errors.password
          && <div className="error">{errors.password}</div>
        }
      </div>
      <div>
        <Label className="label label-block" htmlFor="confirmPassword">
          Confirm Password:
        </Label>
        <Field
          type="password"
          placeholder="Confirm password"
          id="confirmPassword"
          name="confirmPassword"
          required
          className={touched.confirmPassword
          && errors.confirmPassword ? 'input-error' : ''}
        />
        {
          touched.confirmPassword
          && errors.confirmPassword
          && <div className="error">{errors.confirmPassword}</div>
        }
      </div>
      <div>
        <Label className="label label-block" htmlFor="customer">
          Sign up as:
        </Label>
        <Field component="select" name="role" id="role" required>
          <option value="customer">Customer</option>
          <option value="caterer">Caterer</option>
        </Field>
      </div>
      <Button
        value="Sign Up"
        type="submit"
        className="btn btn-default btn-default-lg font-weight-bold"
        disabled={isSubmitting}
      />
      {isSubmitting ?
        <Loading text="signing in">
          <ProgressBar />
        </Loading>
        : null}
    </Form>
  );
};

SignUpForm.propTypes = {
  errors: PropTypes.objectOf(PropTypes.string).isRequired,
  touched: PropTypes.objectOf(PropTypes.bool).isRequired,
  isSubmitting: PropTypes.bool.isRequired,
};

export default SignUpForm;
