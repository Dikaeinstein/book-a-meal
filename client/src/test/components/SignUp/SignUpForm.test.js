import React from 'react';
import { shallow } from 'enzyme';
import SignUpForm from '../../../components/SignUp/SignUpForm';

describe('<SignUpForm />', () => {
  const props = {
    errors: {
      SignUp: '',
      email: '',
      password: '',
      confirmPassword: '',
      role: '',
      name: '',
    },
    touched: {
      email: false,
      password: false,
      name: false,
      confirmPassword: false,
      role: false,
    },
    isSubmitting: false,
  };
  it('should show loading indicator when isSubmitting', () => {
    const tree = shallow(<SignUpForm {...{ ...props, isSubmitting: true }} />);
    expect(tree).toMatchSnapshot();
  });
  it('should show an error if password does not validate', () => {
    const tree = shallow(<SignUpForm {...{
      ...props,
      error: { password: 'error' },
      touched: { password: true },
    }}
    />);
    expect(tree).toMatchSnapshot();
  });
  it('should show an error if email does not validate', () => {
    const tree = shallow(<SignUpForm {...{
      ...props,
      error: { email: 'error' },
      touched: { email: true },
    }}
    />);
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot', () => {
    const tree = shallow(<SignUpForm {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

