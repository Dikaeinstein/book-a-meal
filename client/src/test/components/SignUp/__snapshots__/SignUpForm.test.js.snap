// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> should match snapshot 1`] = `
<Form
  className="form card"
>
  <h5
    className="text-center"
  >
    Sign Up
  </h5>
  <p
    className="text-center font-weight-bold"
  >
    Sign up to book a meal now
  </p>
  <div>
    <Label
      className="label label-block"
      htmlFor="name"
    >
      Full name
    </Label>
    <Field
      className=""
      id="name"
      name="name"
      placeholder="Enter full name"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="email"
    >
      Email address:
    </Label>
    <Field
      className=""
      id="email"
      name="email"
      placeholder="johndoe@domain.com"
      required={true}
      type="email"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="password"
    >
      Password:
    </Label>
    <Field
      className=""
      id="password"
      name="password"
      placeholder="Enter password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="confirmPassword"
    >
      Confirm Password:
    </Label>
    <Field
      className=""
      id="confirmPassword"
      name="confirmPassword"
      placeholder="Confirm password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="customer"
    >
      Sign up as:
    </Label>
    <Field
      component="select"
      id="role"
      name="role"
      required={true}
    >
      <option
        value="customer"
      >
        Customer
      </option>
      <option
        value="caterer"
      >
        Caterer
      </option>
    </Field>
  </div>
  <Button
    className="btn btn-default btn-default-lg font-weight-bold"
    disabled={false}
    type="submit"
    value="Sign Up"
  />
</Form>
`;

exports[`<SignUpForm /> should show an error if email does not validate 1`] = `
<Form
  className="form card"
>
  <h5
    className="text-center"
  >
    Sign Up
  </h5>
  <p
    className="text-center font-weight-bold"
  >
    Sign up to book a meal now
  </p>
  <div>
    <Label
      className="label label-block"
      htmlFor="name"
    >
      Full name
    </Label>
    <Field
      className=""
      id="name"
      name="name"
      placeholder="Enter full name"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="email"
    >
      Email address:
    </Label>
    <Field
      className=""
      id="email"
      name="email"
      placeholder="johndoe@domain.com"
      required={true}
      type="email"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="password"
    >
      Password:
    </Label>
    <Field
      className=""
      id="password"
      name="password"
      placeholder="Enter password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="confirmPassword"
    >
      Confirm Password:
    </Label>
    <Field
      className=""
      id="confirmPassword"
      name="confirmPassword"
      placeholder="Confirm password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="customer"
    >
      Sign up as:
    </Label>
    <Field
      component="select"
      id="role"
      name="role"
      required={true}
    >
      <option
        value="customer"
      >
        Customer
      </option>
      <option
        value="caterer"
      >
        Caterer
      </option>
    </Field>
  </div>
  <Button
    className="btn btn-default btn-default-lg font-weight-bold"
    disabled={false}
    type="submit"
    value="Sign Up"
  />
</Form>
`;

exports[`<SignUpForm /> should show an error if password does not validate 1`] = `
<Form
  className="form card"
>
  <h5
    className="text-center"
  >
    Sign Up
  </h5>
  <p
    className="text-center font-weight-bold"
  >
    Sign up to book a meal now
  </p>
  <div>
    <Label
      className="label label-block"
      htmlFor="name"
    >
      Full name
    </Label>
    <Field
      className=""
      id="name"
      name="name"
      placeholder="Enter full name"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="email"
    >
      Email address:
    </Label>
    <Field
      className=""
      id="email"
      name="email"
      placeholder="johndoe@domain.com"
      required={true}
      type="email"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="password"
    >
      Password:
    </Label>
    <Field
      className=""
      id="password"
      name="password"
      placeholder="Enter password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="confirmPassword"
    >
      Confirm Password:
    </Label>
    <Field
      className=""
      id="confirmPassword"
      name="confirmPassword"
      placeholder="Confirm password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="customer"
    >
      Sign up as:
    </Label>
    <Field
      component="select"
      id="role"
      name="role"
      required={true}
    >
      <option
        value="customer"
      >
        Customer
      </option>
      <option
        value="caterer"
      >
        Caterer
      </option>
    </Field>
  </div>
  <Button
    className="btn btn-default btn-default-lg font-weight-bold"
    disabled={false}
    type="submit"
    value="Sign Up"
  />
</Form>
`;

exports[`<SignUpForm /> should show loading indicator when isSubmitting 1`] = `
<Form
  className="form card"
>
  <h5
    className="text-center"
  >
    Sign Up
  </h5>
  <p
    className="text-center font-weight-bold"
  >
    Sign up to book a meal now
  </p>
  <div>
    <Label
      className="label label-block"
      htmlFor="name"
    >
      Full name
    </Label>
    <Field
      className=""
      id="name"
      name="name"
      placeholder="Enter full name"
      required={true}
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="email"
    >
      Email address:
    </Label>
    <Field
      className=""
      id="email"
      name="email"
      placeholder="johndoe@domain.com"
      required={true}
      type="email"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="password"
    >
      Password:
    </Label>
    <Field
      className=""
      id="password"
      name="password"
      placeholder="Enter password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="confirmPassword"
    >
      Confirm Password:
    </Label>
    <Field
      className=""
      id="confirmPassword"
      name="confirmPassword"
      placeholder="Confirm password"
      required={true}
      type="password"
    />
  </div>
  <div>
    <Label
      className="label label-block"
      htmlFor="customer"
    >
      Sign up as:
    </Label>
    <Field
      component="select"
      id="role"
      name="role"
      required={true}
    >
      <option
        value="customer"
      >
        Customer
      </option>
      <option
        value="caterer"
      >
        Caterer
      </option>
    </Field>
  </div>
  <Button
    className="btn btn-default btn-default-lg font-weight-bold"
    disabled={true}
    type="submit"
    value="Sign Up"
  />
  <Loading
    text="signing in"
  >
    <ProgressBar />
  </Loading>
</Form>
`;
